### PR TITLE
Added target waypoint visualization

### PR DESCRIPTION
**Pull request by S3JER**
Functionality for target waypoints. 

TODO:  Connecting the functionality to the UI

### CODE REVIEW

**Review by henneboy — CHANGES_REQUESTED** [review]
Does this cause trouble for the work in progress by @Casper-NS or @miniJmo ?

**Review by henneboy — APPROVED** [review]

**Comment by mads256h** [comment]
Rebase dis

File: Assets/Scripts/Map/Visualization/Patrolling/PatrollingHeatMapVisualizationMode.cs [hunk 1]
```diff
@@ -19,31 +19,38 @@
...
 
+using Maes.Map;
+using Maes.Map.Visualization.Patrolling;
 using Maes.Statistics;
 
 using UnityEngine;
 
-namespace Maes.Map.Visualization.Patrolling {
-    internal class PatrollingHeatMapVisualizationMode : IPatrollingVisualizationMode {
+namespace Maes.Assets.Scripts.Map.Visualization.Patrolling
```
Comment (mads256h) [inline Assets/Scripts/Map/Visualization/Patrolling/PatrollingHeatMapVisualizationMode.cs:28]: Namespace is fucked up

File: Assets/Scripts/Map/Visualization/Patrolling/PatrollingTargetWaypointVisualizationMode.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+using Maes.Map.Visualization.Patrolling;
+using Maes.PatrollingAlgorithms;
+using Maes.Robot;
+using Maes.Statistics;
+
+namespace Maes.Assets.Scripts.Map.Visualization.Patrolling
```
Comment (mads256h) [inline Assets/Scripts/Map/Visualization/Patrolling/PatrollingTargetWaypointVisualizationMode.cs:6]: Same here

File: Assets/Scripts/Map/Visualization/Patrolling/WaypointHeatMapVisualization.cs [hunk 3]
```diff
@@ -1,7 +1,7 @@
 using Maes.Map.Visualization.Patrolling;
 using Maes.Statistics;
 
-namespace Maes.Map.Visualization
+namespace Maes.Assets.Scripts.Map.Visualization.Patrolling
```
Comment (mads256h) [inline Assets/Scripts/Map/Visualization/Patrolling/WaypointHeatMapVisualization.cs:4]: And here

**Review by mads256h — CHANGES_REQUESTED** [review]

File: Assets/Scripts/UI/SimulationInfoUIControllers/PatrollingInfoUIController.cs [hunk 4]
```diff
@@ -2,7 +2,7 @@
 using System.Collections.Generic;
 
 using Maes.Algorithms;
-using Maes.Map.Visualization;
+using Maes.Assets.Scripts.Map.Visualization.Patrolling;
```
Comment (mads256h) [inline Assets/Scripts/UI/SimulationInfoUIControllers/PatrollingInfoUIController.cs:5]: Can you fix this namespace as well?

**Review by mads256h — CHANGES_REQUESTED** [review]

**Review by mads256h — APPROVED** [review]

**Pull request merged**